src/v/pandaproxy/schema_registry/store.h

```diff
@@ -468,6 +468,46 @@ class store {
         return true;
     }
 
+    ///\brief Get the global mode.
+    result<mode> get_mode() const { return _mode; }
+
+    ///\brief Get the mode for a subject, or fallback to global.
+    result<mode>
+    get_mode(const subject& sub, default_to_global fallback) const {
+        auto sub_it = get_subject_iter(sub, include_deleted::yes);
+        if (sub_it && (sub_it.assume_value())->second.mode.has_value()) {
+            return (sub_it.assume_value())->second.mode.value();
+        } else if (fallback) {
+            return _mode;
+        }
+        return mode_not_found(sub);
```

Comment:
> Should this be `subject_not_found` rather than `mode_not_found`? Should `mode_not_found` be returned if the subject exists and fallback is false?

The mode configuration can exist without a subject. This is required so that a subject can be placed into import without the entire registry being in import.

Reply:
My question is, if `fallback` is `default_to_global::no` and there is no subject `sub`, should we return `subject_not_found` here since there isn't a fallback and there is no subject?

Reply:
>if fallback is default_to_global::no and there is no subject sub, should we return subject_not_found here since there isn't a fallback and there is no subject?

The expected result for `GET /mode/bar` if it hasn't been set is:
```
{"error_code":40409,"message":"Subject 'bar' does not have subject-level mode configured"}
```